components/videoPlayer.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useEffect, useRef, useState } from 'react'
+import 'twin.macro'
+import Loader from './loader'
+import { useInView } from 'react-intersection-observer'
+import PlayButton from './playButton'
+import { VideoMeta } from '../types/post'
+import Social from './social'
+import useFormatter from '../lib/hooks/useFormatter'
+
+interface VideoPlayerProps {
+  poster?: string
+  src: string
+  videoMeta?: VideoMeta
+  likes?: number
+  comments?: number
+  views?: number
+}
+
+const VideoPlayer: FC<VideoPlayerProps> = ({
+  src,
+  poster,
+  videoMeta,
+  likes,
+  comments,
+  views,
+}) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [isPaused, setIsPaused] = useState<boolean>(false)
+  const videoRef = useRef<HTMLVideoElement>(null!)
+  const formatter = useFormatter()
+  const { inView, ref } = useInView({
+    threshold: 0.25,
+  })
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIsPaused(!inView)
+    }
+
+    if (!inView) {
+      videoRef.current.currentTime = 0
+    }
+  }, [inView, isLoading])
+
+  useEffect(() => {
+    if (isPaused) {
+      void videoRef.current.pause()
+    } else {
+      void videoRef.current.play()
+    }
```

Comment:
Я б використала тут тернарний оператор
```suggestion
    isPaused ? 
      void videoRef.current.pause() :   
      void videoRef.current.play() 
```

Reply:
https://eslint.org/docs/rules/no-unused-expressions 
Я использую это дефолтное правило `{allowTernary: false}`, чтобы избежать неясности, как по мне, то использование `if else` очевиднее, чем тернарный оператор. Ознакомься, пожалуйста, с этим правилом и дай фидбек) Но если нужно будет, то я поменяю)

Reply:
я зрозуміла, для мене тернарний оператор здається простішим, ніж `if else`, але розумію що це просто залежить від уподобань розробника, тому залишай `if else`